src/cpu/x64/matmul/amx_blocking_heuristics.hpp

```diff
@@ -0,0 +1,261 @@
+#include "cpu/x64/matmul/brgemm_matmul_utils.hpp"
+//#include "cpu/x64/brgemm/brgemm.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace x64 {
+namespace matmul {
+
+class matmul_amx_blocking_params_t : public brgemm_matmul_conf_t {
+public:
+    matmul_amx_blocking_params_t()
+        : nthr_m_(0)
+        , nthr_n_(0)
+        , nthr_k_(0)
+        , nthr_b_(0)
+        , nthr_mnb_(0)
+        , nthr_(0)
+        , n_blk_(0)
+        , n_chunk_size_(0)
+        , n_chunk_elems_(0)
+        , m_blk_(0)
+        , m_chunk_size_(0)
+        , m_chunk_elems_(0)
+        , k_blk_(0)
+        , k_chunk_size_(0)
+        , k_chunk_elems_(0)
+        , is_a_nt_(true)
+        , is_b_nt_(true)
+        , set_nt_(false)
+        , brgemm_batch_size_(0)
+        , current_lda_(0)
+        , need_buf_c_(false)
+        , need_buf_a_(false)
+        , extendable_k_(false)
+        , blocking_chunk_mem_size_(0)
+        , efficiency_score_(0.0f) {}
+
+    matmul_amx_blocking_params_t(const brgemm_matmul_conf_t &bgmmc)
+        : brgemm_matmul_conf_t(bgmmc)
+        , nthr_m_(nstl::max(nthr_m, 1))
+        , nthr_n_(nstl::max(nthr_n, 1))
+        , nthr_k_(nstl::max(nthr_k, 1))
+        , nthr_b_(nstl::max(nthr_b, 1))
+        , nthr_mnb_(nthr / nthr_k_)
+        , nthr_(nthr_mnb_ * nthr_k_)
+        , n_blk_(N_blk)
+        , n_chunk_size_(N_chunk_size)
+        , n_chunk_elems_(n_blk_ * n_chunk_size_)
+        , m_blk_(M_blk)
+        , m_chunk_size_(M_chunk_size)
+        , m_chunk_elems_(m_blk_ * m_chunk_size_)
+        , k_blk_(K_blk)
+        , k_chunk_size_(K_chunk_size)
+        , k_chunk_elems_(k_blk_ * k_chunk_size_ * brgemm_batch_size)
+        , is_a_nt_(is_a_nt)
+        , is_b_nt_(is_b_nt)
+        , set_nt_(set_nt)
+        , brgemm_batch_size_(brgemm_batch_size)
+        , current_lda_(LDA)
+        , need_buf_c_(use_buffer_c)
+        , need_buf_a_(use_buffer_a)
+        , extendable_k_(extendable_k)
+        , blocking_chunk_mem_size_(0)
+        , efficiency_score_(0.0f) {}
+
+    void update_configuration(brgemm_matmul_conf_t &bgmmc) const;
+    float get_blocking_scores() const { return efficiency_score_; }
+
+    static size_t L1_threshold();
+    static size_t L2_threshold();
+
+protected:
+    virtual float calculate_blocking_scores() = 0;
+    virtual dim_t get_actual_lda();
+
+    // num threads for parallelism wrt k dimension
+    size_t nthr_m_, nthr_n_, nthr_k_, nthr_b_;
+    // num threads for parallelism wrt m, n and batch dimensions
+    int nthr_mnb_;
+    int nthr_;
+    dim_t n_blk_, n_chunk_size_, n_chunk_elems_;
+    dim_t m_blk_, m_chunk_size_, m_chunk_elems_;
+    dim_t k_blk_, k_chunk_size_, k_chunk_elems_;
+
+    bool is_a_nt_, is_b_nt_;
+    bool set_nt_;
+
+    dim_t brgemm_batch_size_;
+    dim_t current_lda_;
+    bool need_buf_c_, need_buf_a_;
+    bool extendable_k_;
+    size_t blocking_chunk_mem_size_;
+    float efficiency_score_;
+
+    bool is_buffer_c_required();
```

Comment:
Consider adding `const` everywhere applicable down below.
```suggestion
    bool is_buffer_c_required() const;
```

Reply:
fixed